### PR TITLE
Аdd restore and migrate functionality

### PR DESCRIPTION
**Pull request by vlvasilev**
**What this PR does / why we need it**:
This PR implements Migrate and Restore function on the Infrastructure Actuator according to PR [#2167](https://github.com/gardener/gardener/pull/2167). 
Migrate deletes the two ConfigMaps which store the Terraform configuration and state. It also deletes the Secret which stores the Terraform variables.
Restore reads the state from Insfrastrucure.Status.State and deploys it in the ConfigMap which store the Terraform state.
**Which issue(s) this PR fixes**:
Part of [#1631](https://github.com/gardener/gardener/issues/1631)

**Special notes for your reviewer**:

**Release note**:

```improvement operator
implement Migrate and Restore functionality
```

### CODE REVIEW

**Comment by rfranzke** [comment]
Converted to a draft PR as its dependencies are not yet resolved. Please mark as 'ready for review' once they are resolved.

**Comment by vlvasilev** [comment]
The PR is tested. But make verify fails with:
```
test/e2e/networkpolicies/networkpolicy_test.go:35:2: SA1019: package github.com/gardener/gardener/test/integration/framework is deprecated: this is the deprecated gardener testframework. Use gardener/test/framework instead  (staticcheck)
        gardenerframework "github.com/gardener/gardener/test/integration/framework"
        ^

```
After merging the [2167](https://github.com/gardener/gardener/pull/2167)
I will remove 
`github.com/gardener/gardener => github.com/vlvasilev/gardener v0.0.0-20200423121540-5ab2b6c74abe` 

At the moment to test it one must only fetch the PR

**Comment by rfranzke** [comment]
Please keep it as "draft PR" if it is still WIP/has dependencies that need to be resolved. Only mark it to be ready as review if it's mergeable.

> The PR is tested. But make verify fails with:

You can add https://github.com/gardener/gardener-extension-provider-aws/pull/73/files#diff-f5e6ed72f74bd5711969dc7bbe35ac73R9-R13

**Comment by rfranzke** [comment]
PR as such looks good :)

**Comment by ialidzhikov** [comment]
@vlvasilev , can you rebase?

File: go.mod [hunk 1]
```diff
@@ -9,13 +9,15 @@ require (
 	github.com/aliyun/aliyun-oss-go-sdk v2.0.1+incompatible
 	github.com/coreos/go-systemd v0.0.0-20190719114852-fd7a80b32e1f
 	github.com/gardener/etcd-druid v0.1.12
-	github.com/gardener/gardener v1.2.1-0.20200408030154-40b97d31d7f7
+	github.com/gardener/gardener v1.3.1-0.20200424112810-331bbbf46108
+	github.com/gardener/gardener-extension-provider-aws v1.6.2 // indirect
```
Comment (rfranzke) [inline go.mod:13]: Why is this here?

Comment (vlvasilev) [reply go.mod]: I have no idea. Thanks

**Comment by ialidzhikov** [comment]
Hmm, `.ci/verify` fails with:
```
> Generate Check
go generate needs to be run:
 M go.mod
make: *** [Makefile:110: check-generate] Error 1
```

@vlvasilev , this repo is still using golang@1.13. Can you try to revendor with golang@v1.13?

File: go.sum [hunk 2]
```diff
@@ -69,6 +69,7 @@ github.com/armon/consul-api v0.0.0-20180202201655-eb2c6b5be1b6/go.mod h1:grANhF5
 github.com/asaskevich/govalidator v0.0.0-20180720115003-f9ffefc3facf/go.mod h1:lB+ZfQJz7igIIfQNfa7Ml4HSf2uFQQRzpGGRXenZAgY=
 github.com/asaskevich/govalidator v0.0.0-20190424111038-f61b66f89f4a/go.mod h1:lB+ZfQJz7igIIfQNfa7Ml4HSf2uFQQRzpGGRXenZAgY=
 github.com/aws/aws-sdk-go v1.13.54/go.mod h1:ZRmQr0FajVIyZ4ZzBYKG5P3ZqPz9IHG41ZoMu1ADI3k=
+github.com/aws/aws-sdk-go v1.19.41 h1:veutzvQP/lOmYmtX26S9mTFJLO6sp7/UsxFcCjglu4A=
```
Comment (rfranzke) [inline go.sum:72]: All these AWS dependencies should not be here, or?

Comment (vlvasilev) [reply go.sum:72]: I have no Idea why those AWS dependencies are in go.sum. Currently they persist in the master branch and after executing `make revendor` they appear again in go.sum. If I search for this dependencies in the project it seems that no one is using them.
